java/ql/src/utils/model-generator/CaptureSinkModels.ql

```diff
@@ -0,0 +1,57 @@
+/**
+ * @name Capture sink models.
+ * @description Finds public methods that act as sinks as they flow into a a known sink.
+ * @id java/utils/model-generator/sink-models
+ */
+
+import java
+private import Telemetry.ExternalAPI
+private import semmle.code.java.dataflow.DataFlow
+private import semmle.code.java.dataflow.TaintTracking
+private import semmle.code.java.dataflow.ExternalFlow
+private import ModelGeneratorUtils
+private import semmle.code.java.dataflow.internal.DataFlowNodes::Private
+
+class PropagateToSinkConfiguration extends TaintTracking::Configuration {
+  PropagateToSinkConfiguration() { this = "parameters or  flowing into sinks" }
+
+  override predicate isSource(DataFlow::Node source) {
+    (source.asExpr() instanceof FieldAccess or source instanceof DataFlow::ParameterNode) and
```

Comment:
The `FieldAccess` gets translated as `Argument[-1]` so we really should verify that the `getFieldQualifier` is a `thisAccess`.